packages/strapi-plugin-graphql/services/Loaders.js

```diff
@@ -85,71 +85,61 @@ module.exports = {
 
       // Generate constant for skip parameters.
       // Note: we shouldn't support both way of doing this kind of things in the future.
-      const skip = query.options.start || query.options.skip;
+      const skip = query.options.start || query.options.skip || 0;
+      const limit = query.options.limit || 100;
 
       // Extracting ids from original request to map with query results.
-      const ids = this.extractIds(query, ref);
-      
-      if (!_.isArray(ids)) {
-        return data
-          .filter(entry => entry[ids.alias].toString() === ids.value.toString())
-          .slice(skip, skip + query.options.limit);
-      }
+      const ids = this.extractIds(query);
+      const ast = ref.associations.find(ast => ast.alias === ids.alias);
+      const astModel = this.retrieveModel(ast.model || ast.collection, ast.plugin);
 
-      // Critical: don't touch this part until you truly understand what you're doing.
-      // The data array takes care of the sorting of the entries. It explains why we are looping from this array and not the `ids` array.
-      // Then, we're applying the `limit`, `start` and `skip` argument.
       return data
         .filter(entry => entry !== undefined)
-        .filter(entry => ids.map(id => id.toString()).includes(entry[ref.primaryKey].toString()))
-        .slice(skip, skip + query.options.limit);
+        .filter(entry => {
+          const aliasEntry = entry[ids.alias];
+          if (_.isArray(aliasEntry)) {
+            return _.find(
+              aliasEntry,
+              value => value[astModel.primaryKey].toString() ===  ids.value
+            );
+          }
+
+          const entryValue = aliasEntry[astModel.primaryKey].toString();
+          return entryValue === ids.value;
+        })
+        .slice(skip, skip + limit);
     });
   },
 
-  extractIds: (query, ref) => {
-    if ( _.get(query.options, `query.${ref.primaryKey}`)) {
-      return  _.get(query.options, `query.${ref.primaryKey}`);
-    }
-
-    // Single object to retrieve (one-to-many).
+  extractIds: (query) => {
     const alias = _.first(Object.keys(query.options.query));
-
+    const value =  query.options.query[alias].toString();
```

Comment:
Please use Lodash... It will break if the alias variable is empty.

Reply:
I think that alias cannot be empty but yeah we may need to use lodash here for security purposes

Reply:
@Aurelsicoko In which case the alias won't be defined? I mean I can use lodash but it's not quite justified here

Reply:
It might happen, we should not take the risk...